code/optimize/src/main/java/com/adobe/marketing/mobile/optimize/Offer.java

```diff
@@ -38,6 +37,7 @@ public class Offer {
     private Map<String, String> characteristics;
 
     SoftReference<OptimizeProposition> propositionReference;
+    PropositionsRepository propositionsRepository = PropositionsRepositoryImpl.INSTANCE;
```

Comment:
Why do we need PropositionsRepository here? Can't we call `OptimizeUtils` methods?

Reply:
updated @spoorthipujariadobe